packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -378,17 +377,28 @@ class VMService {
     }
   }
 
-  Future<Map<String, dynamic>> _handleLoadDartLibrarySources(Map<String,
-      dynamic> data) async {
-    final String kernelPath = data['kernelPath'];
-    final List<int> bytes = await fs.file(kernelPath).readAsBytes();
-    final String result = base64.encode(bytes);
-    return <String, dynamic>{
-      'type': 'Success',
-      'result': <String, String> {
-        'kernelBytes': result,
-      },
-    };
+  Future<Map<String, dynamic>> _handleLoadDartLibrarySources() async {
+    // We assume that the current program is being run using the dart executable
+    // from the dart-sdk packaged with Flutter.
+    const relativeKernelPath =
+      '../../artifacts/engine/common/flutter_patched_sdk/platform_strong.dill';
```

Comment:
I suspect that there's a way to query the flutter_tools api for the artifacts location (possibly via the `Artifacts` class)? cc @jonahwilliams @aam

Reply:
Something like `artifacts.getArtifactPath(Artifact. platformKernelDill)`

Reply:
Good call, totally forgot that existed. Done.